front/app/containers/Admin/projects/edit/general/index.tsx

```diff
@@ -1,576 +0,0 @@
-    }));
-  };
-
-  handleProjectImagesOnAdd = (projectImages: UploadFile[]) => {
-    this.setState({
-      projectImages,
-      submitState: 'enabled',
-    });
-  };
-
-  handleProjectImageOnRemove = (projectImageToRemove: UploadFile) => {
-    this.setState(({ projectImages, projectImagesToRemove }) => ({
-      submitState: 'enabled',
-      projectImages: projectImages.filter(
-        (image) => image.base64 !== projectImageToRemove.base64
-      ),
-      projectImagesToRemove: [...projectImagesToRemove, projectImageToRemove],
-    }));
-  };
-
-  handleTopicsChange = (topicIds: string[]) => {
-    this.setState(({ projectAttributesDiff }) => ({
-      submitState: 'enabled',
-      projectAttributesDiff: {
-        ...projectAttributesDiff,
-        topic_ids: topicIds,
-      },
-    }));
-  };
-
-  handleAreaTypeChange = (value: 'all' | 'selection') => {
-    this.setState(({ projectAttributesDiff }) => ({
-      submitState: 'enabled',
-      areaType: value,
-      projectAttributesDiff: {
-        ...projectAttributesDiff,
-        area_ids: value === 'all' ? [] : projectAttributesDiff.area_ids,
-      },
-    }));
-  };
-
-  handleAreaSelectionChange = (values: IOption[]) => {
-    this.setState(({ projectAttributesDiff }) => ({
-      submitState: 'enabled',
-      projectAttributesDiff: {
-        ...projectAttributesDiff,
-        area_ids: values.map((value) => value.value),
-      },
-    }));
-  };
-
-  onSubmit = (event: FormEvent<any>) => {
-    event.preventDefault();
-
-    const { projectType } = this.state;
-
-    // if it's a new project of type continuous
-    if (projectType === 'continuous') {
-      eventEmitter.emit('getParticipationContext');
-    } else {
-      this.save();
-    }
-  };
-
-  handleParcticipationContextOnSubmit = (
-    participationContextConfig: IParticipationContextConfig
-  ) => {
-    this.save(participationContextConfig);
-  };
-
-  handleStatusChange = (value: PublicationStatus) => {
-    this.setState(({ projectAttributesDiff }) => ({
-      submitState: 'enabled',
-      publicationStatus: value,
-      projectAttributesDiff: {
-        ...projectAttributesDiff,
-        admin_publication_attributes: {
-          publication_status: value,
-        },
-      },
-    }));
-  };
-
-  handleSlugOnChange = (slug: string) => {
-    this.setState(({ projectAttributesDiff }) => {
-      return {
-        slug,
-        projectAttributesDiff: {
-          ...projectAttributesDiff,
-          slug,
-        },
-      };
-    });
-
-    this.validateProjectSlug(slug);
-  };
-
-  validate = () => {
-    const { formatMessage } = this.props.intl;
-    const { currentTenant, projectAttributesDiff, project } = this.state;
-
-    const { hasErrors, titleError } = validate(
-      currentTenant,
-      projectAttributesDiff,
-      project,
-      formatMessage
-    );
-
-    this.setState({
-      titleError: !isEmpty(titleError) ? titleError : null,
-    });
-
-    return !hasErrors;
-  };
```

Comment:
Used equally confusing an imported function (validate) with the same name, which also was a pretty messy function with lots of bad typing. Replaced with validateForm in the new component.